parse.go

```diff
@@ -843,34 +855,45 @@ func (s *scanner) scanString() string {
 		end = s.curr
 	)
 	s.nextChar()
-	i := s.pos - 1
+	i := s.pos - s.currSize
+	if s.currSize > 1 {
+		c++
+	}
 	for s.curr != end {
 		if !s.nextChar() {
 			panic(errors.New("xpath: scanString got unclosed string"))
 		}
-		c++
+		c += s.currSize
 	}
 	s.nextChar()
 	return s.text[i : i+c]
 }
 
 func (s *scanner) scanName() string {
 	var (
-		c int
-		i = s.pos - 1
+		c = s.currSize - 1
+		i = s.pos - s.currSize
 	)
+
+	// Detect current rune size
+
 	for isName(s.curr) {
-		c++
 		if !s.nextChar() {
+			c += s.currSize
 			break
 		}
+		c += s.currSize
 	}
 	return s.text[i : i+c]
 }
 
 func isName(r rune) bool {
 	return string(r) != ":" && string(r) != "/" &&
-		(unicode.Is(first, r) || unicode.Is(second, r) || string(r) == "*")
+		(unicode.Is(first, r) ||
+			unicode.Is(second, r) ||
+			unicode.Is(cyrillic, r) ||
```

Comment:
Why should we need add this new condition for `cyrillic` and `greek`. We already knows Non-English letters yet.

Reply:
Good catch. That was a leftover test from removed code. The idea was to allow certain charsets, but it’s better left to the user to decide post-parsing.

I removed the change